src/services/db/RepoService.ts

```diff
@@ -0,0 +1,247 @@
+    title: string,
+    description?: string
+  ) {
+    return ReviewApi.updatePullRequest(
+      siteName,
+      pullRequestNumber,
+      title,
+      description
+    )
+  }
+
+  closeReviewRequest(siteName: string, pullRequestNumber: number) {
+    return ReviewApi.closeReviewRequest(siteName, pullRequestNumber)
+  }
+
+  mergePullRequest(siteName: string, pullRequestNumber: number) {
+    return ReviewApi.mergePullRequest(siteName, pullRequestNumber)
+  }
+
+  approvePullRequest(siteName: string, pullRequestNumber: number) {
+    return ReviewApi.approvePullRequest(siteName, pullRequestNumber)
+  }
+
+  async getComments(siteName: string, pullRequestNumber: number) {
+    return ReviewApi.getComments(siteName, pullRequestNumber)
+  }
+
+  async createComment(
+    siteName: string,
+    pullRequestNumber: number,
+    user: string,
+    message: string
+  ) {
+    return ReviewApi.createComment(siteName, pullRequestNumber, user, message)
+  }
+
+  getFilePath({ siteName, fileName, directoryName }: any): any {
+    return super.getFilePath({
+      siteName,
+      fileName,
+      directoryName,
+    })
+  }
+
+  getBlobPath({ siteName, fileSha }: any): any {
+    return super.getBlobPath({ siteName, fileSha })
+  }
+
+  getFolderPath({ siteName, directoryName }: any) {
+    return super.getFolderPath({ siteName, directoryName })
+  }
+
+  async create(
+    sessionData: any,
+    { content, fileName, directoryName, isMedia = false }: any
+  ): Promise<any> {
+    return await super.create(sessionData, {
+      content,
+      fileName,
+      directoryName,
+      isMedia,
+    })
+  }
+
+  async read(
+    sessionData: UserWithSiteSessionData,
+    { fileName, directoryName }: { fileName: string; directoryName?: string }
+  ): Promise<GitFile> {
+    if (this.isRepoWhitelisted(sessionData.siteName)) {
+      logger.info("Reading file from local Git file system")
+      const filePath = directoryName ? `${directoryName}/${fileName}` : fileName
+      const result = await this.gitFileSystemService.read(
+        sessionData.siteName,
+        filePath
+      )
+
+      if (result.isErr()) {
+        throw result.error
+      }
+
+      return result.value
+    }
+
+    return await super.read(sessionData, {
+      fileName,
+      directoryName,
+    })
+  }
+
+  // TODO: This is no longer used, remove it
+  async readMedia(sessionData: any, { fileSha }: any): Promise<any> {
+    return await super.readMedia(sessionData, { fileSha })
+  }
+
+  async readDirectory(
+    sessionData: UserWithSiteSessionData,
+    { directoryName }: { directoryName: string }
+  ): Promise<GitDirectoryItem[]> {
+    if (this.isRepoWhitelisted(sessionData.siteName)) {
+      logger.info("Reading directory from local Git file system")
+      const result = await this.gitFileSystemService.listDirectoryContents(
+        sessionData.siteName,
+        directoryName
+      )
+
+      if (result.isErr()) {
+        throw result.error
```

Comment:
do we want to throw errors? in some places we use neverthrow, but some places we throw. is it because of the way upstream calls reposervice?

Reply:
Yup, the way the upstream calls RepoService is that they expect errors to be thrown, so we would have to use this way for now until the various services are updated to handle neverthrow.